config/config/src/main/java/io/helidon/config/internal/UrlOverrideSource.java

```diff
@@ -127,15 +147,26 @@ protected String uid() {
 
         /**
          * Initialize builder.
-         *
-         * @param url configuration url
          */
-        public UrlBuilder(URL url) {
+        private UrlBuilder() {
             super(URL.class);
+        }
 
-            Objects.requireNonNull(url, "url cannot be null");
-
+        /**
+         * Configure the URL that is source of this overrides.
+         *
+         * @param url url of the resource to load
+         * @return updated builder instance
+         */
+        public UrlBuilder url(URL url) {
             this.url = url;
+            return this;
+        }
+
+        @Override
+        public UrlBuilder config(Config metaConfig) {
+            metaConfig.get("url").as(URL.class).ifPresent(this::url);
```

Comment:
I have to say that you are sometimes a bit inconsistent how you use constants. For example in case of KEY_KEY in PrefixedConfigSource you have separate constant for that property but here and on several other places you have the String value written directly to the method.

Reply:
The reason is simple - I use in place text to get stuff from config in general, as that is the only place it is used and the readability of code is actually better without a constant.
The `KEY_KEY` comes from original implementation.

Reply:
Used a constant anyway to be consistent with the rest of the class.